plugins/woocommerce/src/Blocks/BlockTypes/AddToCartForm.php

```diff
@@ -155,6 +138,14 @@ protected function render( $attributes, $content, $block ) {
 		$is_external_product_with_url = $product instanceof \WC_Product_External && $product->get_product_url();
 		$is_stepper_style             = 'stepper' === $attributes['quantitySelectorStyle'] && ! $product->is_sold_individually() && Features::is_enabled( 'add-to-cart-with-options-stepper-layout' );
 
+		add_filter( 'woocommerce_add_to_cart_form_action', function ( $url ) use ( $is_descendent_of_single_product_block ) {
+			if ( $is_descendent_of_single_product_block ) {
```

Comment:
Could this `if` be around the `add_filter()` instead of inside? This way we avoid registering the filter unnecessarily. Ie:

```PHP
if ( $is_descendent_of_single_product_block ) {
  add_filter( ... );
}
```

Also, not a strong opinion, but wondering if we should maybe add a `remove_filter()` after we rendered the template? Just to make sure we leave things as they were before the block was rendered.

Reply:
done -> [d4d15c5](https://github.com/woocommerce/woocommerce/pull/55250/commits/d4d15c595a37300952d4621241d7f8d6f6b50158)